### PR TITLE
RS/YJ/RULE19-24

### PR DESCRIPTION
**Pull request by yunjoonjung-PNNL**
This rule hasn't been tested. I'll test it out when the TCD is available.

### CODE REVIEW

**Review by weilixu — APPROVED** [review]

File: rct229/rulesets/ashrae9012019/section19/section19rule24.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            hvac_p = context.proposed
+            hvac_id_p = hvac_p["id"]
+            inapplicable_hvac_sys_list_p = data["inapplicable_hvac_sys_list_p"]
+
+            return hvac_id_p not in inapplicable_hvac_sys_list_p
+
+        def get_calc_vals(self, context, data=None):
+            hvac_p = context.proposed
+
+            operation_during_unoccupied_p = hvac_p["fan_system"][
```
Comment (JacksonJ-KC) [inline rct229/rulesets/ashrae9012019/section19/section19rule24.py:67]: do we want operation_during_occupied instead?

Comment (yunjoonjung-PNNL) [reply rct229/rulesets/ashrae9012019/section19/section19rule24.py:67]: I'm a bit confused. Don't we get the ```operation_during_unoccupied``` key from the ```fan_system```? So, isn't ```operation_during_unoccupied_p``` correct?

Comment (JacksonJ-KC) [reply rct229/rulesets/ashrae9012019/section19/section19rule24.py:67]: the rule is verifying continuous fan_system operation during occupied hours, not unoccupied hours. RDS also references the operation_during_occupied key

File: rct229/rulesets/ashrae9012019/section19/section19rule24.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        return {"inapplicable_hvac_sys_list_p": inapplicable_hvac_sys_list_p}
+
+    class HVACRule(RuleDefinitionBase):
+        def __init__(self):
+            super(Section19Rule24.HVACRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, False, True),
+                required_fields={
+                    "$": ["fan_system"],
+                    "fan_system": [
+                        "operation_during_occupied",
```
Comment (JacksonJ-KC) [inline rct229/rulesets/ashrae9012019/section19/section19rule24.py:52]: required fields was correctly using `operation_during_occupied` here, but the rest of the code uses `operation_during_unoccupied` when I don't believe it should be.

**Comment by JacksonJ-KC** [comment]
> This rule hasn't been tested. I'll test it out when the TCD is available.

New PR with the 19-24 test RMD [](https://github.com/pnnl/ruleset-checking-tool/pull/1073)

File: rct229/rulesets/ashrae9012019/section19/section19rule24.py (hunk 2 above)
Comment (yunjoonjung-PNNL) [reply rct229/rulesets/ashrae9012019/section19/section19rule24.py:52]: Good catch. Addressed.

File: rct229/rulesets/ashrae9012019/section19/section19rule24.py (hunk 1 above)
Comment (yunjoonjung-PNNL) [reply rct229/rulesets/ashrae9012019/section19/section19rule24.py:67]: Addressed.

**Comment by yunjoonjung-PNNL** [comment]
This rule was tested and passed as expected.

File: docs/section19/Rule19-24.md [hunk 3]
```diff
@@ -1,7 +1,7 @@
-# Airside System – Rule 19-27  
+# Airside System – Rule 19-24  
 **Mandatory Rule:** True  
-**Schema Version:** 0.0.24
-**Rule ID:** 19-27  
+**Schema Version:** 0.0.24   
+**Rule ID:** 19-27
```
Comment (weilixu) [inline docs/section19/Rule19-24.md:4]: Need to update this Rule ID

**Review by weilixu — APPROVED** [review]

**Pull request merged**